spicy/toolchain/src/compiler/validator.cc

```diff
@@ -890,11 +891,18 @@ struct VisitorPost : visitor::PreOrder, hilti::validator::VisitorMixIn {
     }
 
     void operator()(spicy::type::unit::item::UnresolvedField* n) final {
-        if ( auto id = n->unresolvedID() )
-            error(fmt("unknown ID '%s'", id), n, node::ErrorPriority::High);
-        else
-            // I don't think this can actually happen ...
-            error("unit field left unresolved", n);
+        if ( auto id = n->unresolvedID() ) {
+            // Re-lookup ID to see if it exists at all.
+            if ( auto resolved = hilti::scope::lookupID<hilti::Declaration>(id, n, "field"); ! resolved )
+                error(resolved.error(), n, node::ErrorPriority::High);
+
+            if ( n->hasErrors() )
+                // Report existing error, probably from the resolver.
+                return;
+        }
+
+        // I don't think this can actually happen ...
```

Comment:
Little nit: Would an assertion here make sense, just so a debug build would show if this case needs a bit more care? Now it's relying on `n->hasErrors()` as well, so other changes could technically cause this to have weird behavior that'd be nice for an assertion to catch IMO

Reply:
Let me make it an internal error in both release/debug, then we'll at least see for sure if/when it happens unexpectedly.